includes/api/class-wc-rest-products-controller.php

```diff
@@ -1702,8 +1704,7 @@ protected function update_post_meta_fields( $post, $request ) {
 					$this->save_variations_data( $product, $request );
 				} else {
 					// Just sync variations.
-					WC_Product_Variable::sync( $product->get_id() );
-					WC_Product_Variable::sync_stock_status( $product->get_id() );
+					WC_Product_Variable::sync( $product, true );
```

Comment:
@justinshreve after your data store change, is this correct? Passing an ID should still work.